reflect/src/main/java/dagger/reflect/ReflectiveModuleParser.java

```diff
@@ -54,7 +54,7 @@ static void parse(Class<?> moduleClass, @Nullable Object instance,
             continue;
           }
         } else {
-          if ((method.getModifiers() & STATIC) == 0 && instance == null) {
+          if ((method.getModifiers() & STATIC) == 0 && instance == null && !moduleClass.isInterface()) {
```

Comment:
with this added condition it lets too much through, and throws NPE at linking time
without this condition it check unrelated methods, and throws a misleading message

Reply:
Initialize `binding` to `null`, check if `DEFAULT` in modifiers, throw nice message if `binding != null` otherwise ignore?

Reply:
Outdated, resolving.
Note: there's no `Modifier.isDefault(method.getModifiers())` only `method.isDefault()`.